subprojects/testing-base/src/testFixtures/groovy/org/gradle/testing/AbstractTestFrameworkIntegrationTest.groovy

```diff
@@ -150,11 +157,114 @@ abstract class AbstractTestFrameworkIntegrationTest extends AbstractIntegrationS
+        when:
+        buildFile << "${testTaskName}.filter.includeTestsMatching '${testSuite('SomeTest')}.missingMethod'"
+        fails(testTaskName)
+        then: failure.assertHasCause("No tests found for given includes: [${testSuite('SomeTest')}.missingMethod](filter.includeTestsMatching)")
+    }
+
+    def "task is out of date when --tests argument changes"() {
+        String testTaskPath = ":$testTaskName"
+
+        given:
+        createPassingFailingTest()
+
+        when: run(testTaskName, "--tests", "${testSuite('SomeOtherTest')}.$passingTestCaseName")
+        then: testResult.testClass("SomeOtherTest").assertTestsExecuted(passingTestCaseName)
+
+        when: run(testTaskName, "--tests", "${testSuite('SomeOtherTest')}.$passingTestCaseName")
+        then: result.skippedTasks.contains(testTaskPath) //up-to-date
+
+        when:
+        run(testTaskName, "--tests", "${testSuite('SomeTest')}.$passingTestCaseName")
+
+        then:
+        !result.skippedTasks.contains(testTaskPath)
+        testResult.testClass("SomeTest").assertTestsExecuted(passingTestCaseName)
+    }
+
+    @Unroll
+    def "can select multiple tests from commandline #scenario"() {
+        given:
+        createPassingFailingTest()
+
+        when:
+        runAndFail(stringArrayOf(command))
+
+        then:
+        testResult.assertTestClassesExecuted(stringArrayOf(classesExecuted))
+        if (!someOtherTestExecuted.isEmpty()) {
+            testResult.testClass("SomeOtherTest").assertTestsExecuted(stringArrayOf(someOtherTestExecuted))
+        }
+        testResult.testClass("SomeTest").assertTestsExecuted(stringArrayOf(someTestExecuted))
+
+        where:
+        scenario                      | command                                                                                                                                                       | classesExecuted               | someOtherTestExecuted | someTestExecuted
+        "no options"                  | [testTaskName]                                                                                                                                                | ["SomeTest", "SomeOtherTest"] | [passingTestCaseName] | [passingTestCaseName, failingTestCaseName]
+        "fail and SomeTest.pass"      | [testTaskName, "--tests", "${testSuite('SomeTest')}.$failingTestCaseName", "--tests", "${testSuite('SomeTest')}.$passingTestCaseName"]      | ["SomeTest"]                  | []                    | [passingTestCaseName, failingTestCaseName]
+        "fail and SomeOtherTest.fail" | [testTaskName, "--tests", "${testSuite('SomeTest')}.$failingTestCaseName", "--tests", "${testSuite('SomeOtherTest')}.$passingTestCaseName"] | ["SomeTest", "SomeOtherTest"] | [passingTestCaseName] | [failingTestCaseName]
+    }
+
+    @Unroll
+    def "can deduplicate test filters when #scenario"() {
+        given:
+        createPassingFailingTest()
+
+        when:
+        runAndFail(stringArrayOf(command))
```

Comment:
I think you can avoid needing `stringArrayOf` by using `*command` or by declaring `String[] command` as an argument to the method.

Reply:
Done.